src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -90,20 +93,25 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 		this._gutterDecorationOpaque.glyphMarginHoverMessage = hoverMessage;
 	}
 
-	private _onEnablementOrModelChanged(): void {
+	private _updateCurrentBreakpoints(uri: URI) {
+		this._currentBreakpoints = this._debugService.getModel().getBreakpoints({ uri });
+	}
+
+	private _onEnablementOrModelChanged(executeOnlyDisposal: boolean = false): void {
```

Comment:
But also `executeOnlyDisposal` is not nice. 
What if it takes a while to get a session started and events come in this order:
- `onWillStartSession`: _onEnablementOrModelChanged will dispose
- a provider is added: _onEnablementOrModelChanged calls this._inlineChatSessionService.getSession but it is still undefied
- now the session is up, but there's no event

Reply:
Because the only event we have is `onWillStartSession` and `onOnDidSession` we can't use `getSession`, but need to remember it ourselves

Reply:
I think executeOnlyDisposal is perhaps not so nice indeed, but it was added to centralize the code as per the review. I see better, I thought you meant previously we should use directly the inline chat session service for that, but I'll add a local property.

Reply:
Now you can remove `executeOnlyDisposal`